### ISSUE TITLE
inNode HomeKit out - auto start 

### ISSUE THREAD

**Issue by TheNON75** [deconz-community/node-red-contrib-deconz#121]
Dear All,

I would like to recommend to implement the possibility of switching off the sending of payload on the HomeKit output on start up/restart.
for example if the in node is dimmer and the last message was on:true then the light or the controlled accessory will be turned on or toggled, depending on how the payload was used.
The current method may cause issues, when performing a maintenance or simply performing a full deployment during the night as some lights or other accessories (such as speakers) may turn on based on the dimmer.
Typically this happens in the following cases:

- the dimmer (or any kind of switch) is acting as a toggle on/off
- there are more dimmers (switches) that are controlling an accessory, therefore one of the actors' last message will be with high probability on:true
- the accessories are controlled also from an application such as home app

thank you for considering

**Comment by Zehir** [comment]
You mean like the Start Output 1 to toogle or not the send state on start ?

**Comment by TheNON75** [comment]
Exactly

**Comment by djiwondee** [comment]
I would also vote for this. Have similar/same experience. Controlling two light bulbs in my bedroom with the inNode and there are some nights the lights switches on while we are sleeping. Was wondering about that, also after partly or full deployment the lights switch always to on.

**Comment by Zehir** [comment]
This will be part of #120 but will take sime days to come

**Comment by Zehir** [comment]
> I would also vote for this. Have similar/same experience. Controlling two light bulbs in my bedroom with the inNode and there are some nights the lights switches on while we are sleeping. Was wondering about that, also after partly or full deployment the lights switch always to on.

Are you sure that is not because power outage ?

**Comment by djiwondee** [comment]
> Are you sure that is not because power outage ?

No, pretty sure! This is repeatable traceable. Power outage would hove more impacts on other devices in my setup as well (e. g. Windows blinds which would go up as per default)

**Comment by Zehir** [comment]
> > Are you sure that is not because power outage ?
> 
> No, pretty sure! This is repeatable traceable. Power outage would hove more impacts on other devices in my setup as well (e. g. Windows blinds which would go up as per default)

OK i see the problem, is was  introduced since I take the developpement of it or it was already the case before ?

It's because on start and on websocket reconnexion he send the current state but for buttons events they are sent to the your flow think there was an button event but no.
So you will have the same problem if you have connexion lost between node-red and deconz

**Comment by Zehir** [comment]
If you set the `Output` to `on update` this will stop the problem on reconnect and the [new checkbox](https://github.com/deconz-community/node-red-contrib-deconz/pull/120#issuecomment-791964920) to send on start will solve this issue.

**Comment by Zehir** [comment]
The outputAtStartup settings of both for on start and on reconnect so you don't need to set `on update`

**Comment by Zehir** [comment]
@djiwondee until the release of #120 you can parse the value of `meta.state.lastupdated` and check if it's within 30 seconds from now.

**Comment by djiwondee** [comment]
@Zehir 

> The outputAtStartup settings of both for on start and on reconnect so you don't need to set `on update`

Can you please state more clear what you mean? This is how it looks like today:

<img width="798" alt="image" src="https://user-images.githubusercontent.com/37173958/110246299-64892280-7f67-11eb-850c-af015a681d68.png">

<img width="473" alt="image" src="https://user-images.githubusercontent.com/37173958/110246338-897d9580-7f67-11eb-91bf-f81ae499a74f.png">

If I connect follow up nodes to the output #2 the button press event is not passed through.

**Comment by Zehir** [comment]
The output 1 send button press event but not in a format comprehensible for homekit nodes.
The problem is when node-red start and when the nodes reconnect to deconz after a connexion lost, he send the last button event again. That why there is a setting on output 1 but there is not on output 2. The setting disable the re-send of the last button event.
